rust/arrow/src/array/builder.rs

```diff
@@ -3969,579 +3296,4 @@ mod tests {
         // Special error if the key overflows (256th entry)
         builder.append(1257).unwrap();
     }
-
-    #[test]
-    fn test_primitive_append() -> Result<()> {
```

Comment:
is there value in porting some of these edge-case tests for mutable array?

Reply:
Yes, there is. PR's description:

> I would like to migrate all the tests for the XBuilder::append_data to the MutableArrayData, to not lose them, but for that #8850 #8852 #8851 and #8849 and #8848 needs to land first (thus being a draft).

;)

Reply:
Ha, my bad, my eyes couldn't resist to skip that part and went into the bechmark section :P

Reply:
I have incorporated all tests. There were some small bugs in the tests wrt to nulls, and also a bug in the `MutableDataArray`. So, overall, everyone won there.